config/config/src/main/java/io/helidon/config/ConfigSourcesRuntime.java

```diff
@@ -0,0 +1,223 @@
+        ConfigKeyImpl key = ConfigKeyImpl.of(changedKey);
+        ObjectNode changeObjectNode = toObjectNode(changeNode);
+
+        if (key.isRoot()) {
+            // we have a root, no merging with original data, just return it
+            return Optional.of(changeObjectNode);
+        }
+
+        ObjectNode newRootNode = ObjectNode.builder().addObject(changedKey, changeObjectNode).build();
+
+        // old data was empty, this is the only data we have
+        if (oldData.isEmpty()) {
+            return Optional.of(newRootNode);
+        }
+
+        // we had data, now we have new data (not on root), let's merge
+        return Optional.of(mergingStrategy.merge(List.of(newRootNode, oldData.get())));
+    }
+
+    private ObjectNode toObjectNode(ConfigNode changeNode) {
+        switch (changeNode.nodeType()) {
+        case OBJECT:
+            return (ObjectNode) changeNode;
+        case LIST:
+            return ObjectNode.builder().addList("", (ConfigNode.ListNode) changeNode).build();
+        case VALUE:
+            return ObjectNode.builder().value(((ConfigNode.ValueNode) changeNode).get()).build();
+        default:
+            throw new IllegalArgumentException("Unsupported node type: " + changeNode.nodeType());
+        }
+    }
+
+    synchronized Optional<ObjectNode> latest() {
+        List<ObjectNode> objectNodes = loadedData.stream()
+                .map(RuntimeWithData::data)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return Optional.of(mergingStrategy.merge(objectNodes));
+    }
+
+    synchronized Optional<ObjectNode> load() {
+
+        for (ConfigSourceRuntimeBase source : allSources) {
+            if (source.isLazy()) {
+                loadedData.add(new RuntimeWithData(source, Optional.empty()));
+            } else {
+                loadedData.add(new RuntimeWithData(source, source.load()
+                        .map(ObjectNodeImpl::wrap)
+                        .map(objectNode -> objectNode.initDescription(source.description()))));
+            }
+        }
+
+        Set<String> allKeys = loadedData.stream()
+                .map(RuntimeWithData::data)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .flatMap(this::streamKeys)
+                .collect(Collectors.toSet());
+
+        if (allKeys.isEmpty()) {
+            return Optional.empty();
+        }
+
+        // now we have all the keys, let's load them from the lazy sources
+        for (RuntimeWithData data : loadedData) {
+            if (data.runtime().isLazy()) {
+                data.data(loadLazy(data.runtime(), allKeys));
+            }
+        }
+
+        List<ObjectNode> objectNodes = loadedData.stream()
+                .map(RuntimeWithData::data)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
```

Comment:
same

Reply:
Fixing

Reply:
Also fixing in all other classes this was used.